staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -138,17 +144,34 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (o
 	return object, nil
 }
 
+// Returns true if the managedFields indicate that the user is trying to
+// reset the managedFields, i.e. if the list is non-nil but empty, or if
+// the list has one empty item.
+func isResetManagedFields(managedFields []metav1.ManagedFieldsEntry) bool {
+	if len(managedFields) == 0 {
+		return managedFields != nil
+	}
+
+	if len(managedFields) == 1 {
+		mf := managedFields[0]
```

Comment:
why not `reflect.DeepEqual(managedFields[0], metav1.ManagedFieldsEntry{})`?

Reply:
Done

Reply:
did this get pushed?

Reply:
I must have been interrupted somehow? Sorry about that ...